File: 3. Codigo/PROYECTO_ADSFW/src/main/java/com/bocatto/model/Menu.java
package com.bocatto.model;

import java.math.BigDecimal;

public class Menu {

    private String id;
    private String nombre;
    private String descripcion;
    private BigDecimal precio;
    private String categoria;
    private int disponibilidad;

    public Menu(String id, String nombre, String descripcion, BigDecimal precio, String categoria, int disponibilidad) {
        this.id = id;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.categoria = categoria;
        this.disponibilidad = disponibilidad;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public BigDecimal getPrecio() {
        return precio;
    }

    public void setPrecio(BigDecimal precio) {
        this.precio = precio;
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public int getDisponibilidad() {
        return disponibilidad;
    }

    public void setDisponibilidad(int disponibilidad) {
        this.disponibilidad = disponibilidad;
    }

    @Override
    public String toString() {
        return nombre + " - $" + precio;
    }
}

File: 3. Codigo/BOCATTO_V2.0/backend/src/models/Menu.js
import mongoose from "mongoose";

const menuSchema = new mongoose.Schema(
    {
        nombre: {
            type: String,
            required: true
        },

        descripcion: String,

        categoria: String,

        precio: {
            type: Number,
            required: true
        },

        imagen: {
            type: String,
            default: ""
        },

        disponible: {
            type: Boolean,
            default: true
        }
    },
    {
        timestamps: true,
        versionKey: false
    }
);

export default mongoose.model("Menu", menuSchema);